objects/PlayerObject.object.gmx

```diff
@@ -66,10 +66,14 @@ keyDown = keyboard_check(ord("S"))
 hsp = playerSpeed * ((-1 * keyLeft) + keyRight)
 vsp = playerSpeed * ((-1 * keyUp) + keyDown)
 
-if hsp != 0 &amp;&amp; vsp != 0 {
-    hsp = round(hsp/sqrt(2))
-    vsp = round(vsp/sqrt(2))
-}
+// In case hsp and vsp are ever not boolean on/off
+mag = sqrt(hsp*hsp + vsp+vsp)
+hsp /= mag
+vsp /= mag
```

Comment:
These variables should be rounded, because GameMaker is a bit wonky and will try to move the player fractions of a pixel causing the player sprite to become a blurry mess.